Tasks/Common/azure-arm-rest/azure-arm-common.ts

```diff
@@ -67,9 +82,55 @@ export class ApplicationTokenCredentials {
         return this.clientId;
     }
 
-    private getAuthorizationToken(): Q.Promise<string> {
+    private _getMSIAuthorizationToken(count: number ,timeToWait: number): Q.Promise<string> {
         var deferred = Q.defer<string>();
+        let webRequest = new webClient.WebRequest();
+        webRequest.method = "GET";
+        let apiVersion = "2018-02-01";
+        const retryLimit = 5;
+        let msiClientId =  this.msiClientId ? "&client_id=" + this.msiClientId : "";       
+        webRequest.uri = "http://169.254.169.254/metadata/identity/oauth2/token?api-version=" + apiVersion + "&resource="+ this.baseUrl + msiClientId;
+        webRequest.headers = {
+            "Metadata": true
+        };
+
+        webClient.sendRequest(webRequest).then(
+            (response: webClient.WebResponse) => {
+                if (response.statusCode == 200) 
+                {
+                    deferred.resolve(response.body.access_token);
+                }
+                else if (response.statusCode == 209 || response.statusCode == 500)
```

Comment:
should we[ retry for 429](https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service#error) instead of 209?